src/places/places.service.ts

```diff
@@ -38,77 +38,88 @@ export class PlacesService extends createPrismaBase(MODELS.Place) {
       ...(slug ? { slug } : {}),
     }
 
-    let places: (
-      | Prisma.PlaceGetPayload<{ select: Prisma.PlaceSelect }>
-      | Prisma.PlaceGetPayload<{ include: Prisma.PlaceInclude }>
-    )[]
+    const placeSelectBase: Prisma.PlaceSelect | undefined = placeColumns
+      ? (buildColumnSelect(placeColumns) as Prisma.PlaceSelect)
+      : undefined
 
     const buildRaceInclude = () => {
-      if (raceColumns) {
-        const raceSelect = buildColumnSelect(raceColumns) as Prisma.RaceSelect
-        return { select: raceSelect }
-      } else {
-        return true
+      if (!raceColumns) return true
+      if (!includeRaces) return true
+
+      // Force add slug and positionNames so we can dedupe by slug
+      const cols = Array.from(
+        new Set([...raceColumns.split(','), 'slug', 'positionNames']),
```

Comment:
Prolly good to put these string literals into constants like `SLUG_COLUMN_NAME` and `POSITION_NAMES_COLUMN_NAME` so that if in the future these change, they'll be available as a single point of truth